ckanext/xloader/jobs.py

```diff
@@ -237,9 +237,40 @@ def _download_resource_data(resource, data, api_key, logger):
     data['datastore_contains_all_records_of_source_file'] = False
     which will be saved to the resource later on.
     '''
-    # check scheme
     url = resource.get('url')
-    scheme = urlparse.urlsplit(url).scheme
+    url_parse = urlparse.urlsplit(url)
+
+    # check if it is an uploaded file
+    domain = url_parse.netloc
+    site_url = config.get('ckan.site_url')
+    if resource.get('url_type') != 'upload' and domain != site_url:
+        raise JobError('Only uploaded files can be added to the Data Store.')
+
+    # get url from cloudstorage
+    filename = url_parse.path.split('/')[-1]
+    from ckanext.cloudstorage.storage import ResourceCloudStorage
+    from ckan import logic, model
+    from pylons import c
+
+    context = {
+        'model': model,
+        'session': model.Session,
+        'user': c.user or c.author,
+        'auth_user_obj': c.userobj
+    }
```

Comment:
is this necessary? I'm pretty sure an empty context will be auto-filled with this stuff, e.g. https://github.com/ckan/ckanapi/blob/master/ckanapi/localckan.py#L26 just passes `{}` by default